src/ts/message.ts

```diff
@@ -104,7 +108,20 @@ export function printMessage(fileName: string, exportMap: ExportMap, messageDesc
         exportType = filePathToPseudoNamespace(fieldEnumType.fileName) + "." + withinNamespace;
       }
     } else {
-      exportType = getTypeName(type);
+      if (field.getOptions() && field.getOptions().hasJstype()) {
```

Comment:
Sorry for making this code even more disgusting, I do have a local branch which contains a large refactor to message parsing - I'd say it's about 75% complete .

Reply:
Done, thanks for the suggestion.